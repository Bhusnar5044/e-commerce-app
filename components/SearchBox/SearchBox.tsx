'use client';

import { usePathname, useRouter } from 'next/navigation';
import { ChangeEvent, FC, KeyboardEvent, memo, useCallback, useState } from 'react';

export const SearchBox: FC = memo(() => {
    const [input, setInput] = useState('');
    const router = useRouter();
    const pathname = usePathname();

    const handleInputChanges = useCallback((event: ChangeEvent<HTMLInputElement>) => {
        setInput(event.target.value);
    }, []);

    const handleEnterKey = useCallback(
        (event: KeyboardEvent<HTMLInputElement>) => {
            if (event.key === 'Enter') router?.push(`${pathname}?search=${input}`);
        },
        [input, pathname, router],
    );

    const handleSubmit = useCallback(() => {
        router?.push(`${pathname}?search=${input}`);
    }, [input, pathname, router]);

    return (
        <>
            <label htmlFor="default-search" className="mb-2 text-sm font-medium text-gray-900 sr-only dark:text-white">
                Search
            </label>
            <div className="relative">
                <input
                    type="search"
                    id="default-search"
                    className="block w-full p-2 pl-4 text-sm text-gray-900 border border-gray-300 rounded-lg focus:ring-blue-500 focus:border-blue-500  dark:border-gray-600 dark:placeholder-gray-400 dark:focus:ring-blue-500 dark:focus:border-blue-500"
                    placeholder="Search Mockups, Logos..."
                    onChange={handleInputChanges}
                    value={input}
                    onKeyDown={handleEnterKey}
                    required
                />
                <div
                    onClick={handleSubmit}
                    className="absolute inset-y-1 right-0 flex items-center pr-3 pointer-events-cursor"
                >
                    <svg
                        className="w-4 h-4 text-gray-500 dark:text-gray-400"
                        aria-hidden="true"
                        xmlns="http://www.w3.org/2000/svg"
                        fill="none"
                        viewBox="0 0 20 20"
                    >
                        <path
                            stroke="currentColor"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                            d="m19 19-4-4m0-7A7 7 0 1 1 1 8a7 7 0 0 1 14 0Z"
                        />
                    </svg>
                </div>
            </div>
        </>
    );
});

SearchBox.displayName = 'SearchBox';
