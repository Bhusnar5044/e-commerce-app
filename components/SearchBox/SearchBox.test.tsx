import { render } from '@test-utils';
import { SearchBox } from './SearchBox';

describe('SearchBox', () => {
    it('should render properly', () => {
        const { container } = render(<SearchBox />);
        expect(container).toMatchSnapshot();
    });
});
