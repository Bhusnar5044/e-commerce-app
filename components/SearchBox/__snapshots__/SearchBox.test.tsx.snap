// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox should render properly 1`] = `
<div>
  <label
    class="mb-2 text-sm font-medium text-gray-900 sr-only dark:text-white"
    for="default-search"
  >
    Search
  </label>
  <div
    class="relative"
  >
    <input
      class="block w-full p-2 pl-4 text-sm text-gray-900 border border-gray-300 rounded-lg focus:ring-blue-500 focus:border-blue-500  dark:border-gray-600 dark:placeholder-gray-400 dark:focus:ring-blue-500 dark:focus:border-blue-500"
      id="default-search"
      placeholder="Search Mockups, Logos..."
      required=""
      type="search"
      value=""
    />
    <div
      class="absolute inset-y-1 right-0 flex items-center pr-3 pointer-events-cursor"
    >
      <svg
        aria-hidden="true"
        class="w-4 h-4 text-gray-500 dark:text-gray-400"
        fill="none"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m19 19-4-4m0-7A7 7 0 1 1 1 8a7 7 0 0 1 14 0Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </div>
</div>
`;
